cmd/initialise.go

```diff
@@ -0,0 +1,189 @@
+// Copyright © 2019 Zappi
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cmd
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/kms"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"github.com/spf13/cobra"
+)
+
+var httpClient    http.Client
+
+// InitPayload holds a Vault init request.
+type InitPayload struct {
+	RecoveryShares    int `json:"recovery_shares"`
+	RecoveryThreshold int `json:"recovery_threshold"`
+	SecretShares      int `json:"secret_shares"`
+	SecretThreshold   int `json:"secret_threshold"`
+}
+
+// InitResponse holds a Vault init response.
+type InitResponse struct {
+	Keys               []string `json:"keys"`
+	KeysBase64         []string `json:"keys_base64"`
+	RootToken          string   `json:"root_token"`
+	RecoveryKeys       []string `json:"recovery_keys"`
+	RecoveryKeysBase64 []string `json:"recovery_keys_base64"`
+}
+
+// initialiseCmd represents the initialise command
+var initialiseCmd = &cobra.Command{
+	Use:   "initialise",
+	Short: "A brief description of your command",
```

Comment:
This needs to be updated.

Reply:
Updated in https://github.com/Intellection/vault-init/pull/1/commits/282eff9e6cb1740337fa647d1fdb9102b7369054